tests/frame/with_columns_test.py

```diff
@@ -18,18 +18,25 @@ def test_with_columns_int_col_name_pandas() -> None:
     pd.testing.assert_frame_equal(result, expected)
 
 
-def test_with_columns_order(constructor_eager: Any) -> None:
+def test_with_columns_order(constructor: Any) -> None:
     data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
-    df = nw.from_native(constructor_eager(data))
+    df = nw.from_native(constructor(data))
     result = df.with_columns(nw.col("a") + 1, d=nw.col("a") - 1)
-    assert result.columns == ["a", "b", "z", "d"]
+    assert result.collect_schema().names() == ["a", "b", "z", "d"]
     expected = {"a": [2, 4, 3], "b": [4, 4, 6], "z": [7.0, 8, 9], "d": [0, 2, 1]}
     compare_dicts(result, expected)
 
 
+def test_with_columns_empty(constructor: Any) -> None:
+    data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
+    df = nw.from_native(constructor(data))
+    result = df.select().with_columns()
+    compare_dicts(result, {})
+
+
 def test_with_columns_order_single_row(constructor_eager: Any) -> None:
     data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
-    df = nw.from_native(constructor_eager(data)[:1])
+    df = nw.from_native(constructor_eager(data)[:1], eager_only=True)
     assert len(df) == 1
```

Comment:
If we skip the assertion on `len(df)==1` (which is kind of assessed anyway as expected has lists of len 1, we can use `constructor` as:

```py
    df = nw.from_native(constructor(data)).head(1)
```

and mark it fail for dask (as we don't implement head)